Update reference docs for reindex resilience (#148786)
elastic/elasticsearch: docs/reference/elasticsearch/rest-apis/reindex-indices.md

Reviewer: Can we change "sliced scroll" to "sliced search" here, just to be pedantic. (I know the section heading we're linking to say "scroll", but it does talk about both, so let's avoid the risk of spreading misinformation here.)

- [Reindex in {{cps}}](#reindex-cps)

**Troubleshooting**

> - [Monitor asynchronous reindex operations](#monitor-reindex-tasks)

Author: you didn't add it, but wonder whether `manage` would be a better word than `monitor`, monitor makes it sound like you're just watching it rather than cancelling/rethrottling

same for other usages of monitor in this PR

Reviewer: Yep, that's a good call. I'm leaving the anchor unchanged, since folks might have links or bookmarks into this, but I'll change the word.

Reviewer: FYI, this section doesn't cover rethrottle, as that already has a section of its own. But it does include cancel. Maybe I'll just change the title to say 'monitor and cancel'. And I'll update the other references according to context.

Reviewer: Meh, no, actually I'm gonna go with 'manage' as you suggest, having looked at the other contexts...

## Reindex asynchronously [docs-reindex-task-api]

> If the request contains `wait_for_completion=false`, {{es}} performs some preflight checks, launches the request, and returns an ID you can use to [monitor](#monitor-reindex-tasks) the operation.

Author: nit/preference
```suggestion
If the request contains `wait_for_completion=false`, {{es}} performs some preflight checks, launches the request, and returns a task ID you can use to [monitor](#monitor-reindex-tasks) the operation.
```

## Reindex asynchronously [docs-reindex-task-api]

If the request contains `wait_for_completion=false`, {{es}} performs some preflight checks, launches the request, and returns a task ID you can use to [manage](#monitor-reindex-tasks) the operation.

> For long-running reindexes, prefer async reindexes. Synchronous reindex keeps a client waiting on the node that received the request, and this will time out.

Reviewer 2: ```suggestion
For long-running reindexes, prefer async reindexes. Synchronous reindex keeps a client waiting on the node that received the request and this will time out.
```

Set `requests_per_second` to any positive decimal number (for example, `1.4`, `6`, or `1000`) to throttle the rate at which the reindex API issues batches of index operations.
Requests are throttled by padding each batch with a wait time.
To turn off throttling, set `requests_per_second` to `-1`.

The throttling is done by waiting between batches.

> Set the underlying search keep-alive long enough that a slower batch does not expire the context before the next read (see the following note).

Reviewer 2: We generally avoid "see" and I think that phrase is unnecessary anyway
```suggestion
Set the underlying search keep-alive long enough that a slower batch does not expire the context before the next read.
```

> :::::{note}

Reviewer 2: I think it's simpler to just use a plain admonition per https://elastic.github.io/docs-builder/syntax/admonitions/#plain
```suggestion
:::::{admonition} Scroll query parameter versus PIT keep-alive
```

> **`scroll` query parameter versus PIT keep-alive**

Reviewer 2: The title can be moved up into the admonition declaration
```suggestion

```

::::{applies-switch}

:::{applies-item} { "stack": "ga 9.5+", "serverless": "ga" }

> Reindex reads the source with **point-in-time** pagination for local reindexes, and for reindex from remote when the remote cluster is **{{es}} 7.10 or later** (so a PIT can be opened there). The top-level **`scroll` parameter on the reindex request has no effect** on that path. Reindex from a remote cluster **older than {{es}} 7.10** cannot use the PIT path and **falls back to scroll**; the **`scroll`** parameter then sets scroll keep-alive (not `cluster.reindex.pit.keep_alive`). Use [`cluster.reindex.pit.keep_alive`](/reference/elasticsearch/configuration-reference/index-management-settings.md#reindex-settings) to change how long those contexts stay open.

Reviewer 2: I'm personally not a fan of a lot of bolding, particularly since our style guide mainly uses it for UI elements per https://www.elastic.co/docs/contribute-docs/style-guide/formatting#bold-strong
```suggestion
Reindex reads the source with point-in-time pagination for local reindexes and for reindex from remote when the remote cluster is {{es}} 7.10 or later (so a PIT can be opened there). The top-level `scroll` parameter on the reindex request has no effect on that path.

Reindex from a remote cluster older than {{es}} 7.10 cannot use the PIT path and falls back to scroll; the `scroll` parameter then sets scroll keep-alive (not `cluster.reindex.pit.keep_alive`). Use [`cluster.reindex.pit.keep_alive`](/reference/elasticsearch/configuration-reference/index-management-settings.md#reindex-settings) to change how long those contexts stay open.
```

Reviewer: Huh, I am sure I removed all the bolds from this paragraph once already, but it seems that didn't quite make it into the commit I actually pushed. I'm not sure how that happened. Anyway, I agree.

:::

:::{applies-item} { "stack": "ga 9.0-9.4" }

> Reindex uses **scroll**-based pagination for local and remote sources. The **`scroll`** parameter sets scroll keep-alive; allow enough time for throttling gaps between batches.

Reviewer 2: Ditto re bolding:
```suggestion
Reindex uses scroll-based pagination for local and remote sources. The `scroll` parameter sets scroll keep-alive; allow enough time for throttling gaps between batches.
```

POST _reindex/r1A2WoRbTwKZ516z6NEs5A:36619/_rethrottle?requests_per_second=-1
```

> Use the `task` ID returned from the call to `POST _reindex`, or find it using the [management](#monitor-reindex-tasks) APIs.

Reviewer 2: ```suggestion
Use the `task` ID returned from the call to `POST _reindex` or find it using the [management](#monitor-reindex-tasks) APIs.
```

> Similarly to setting `requests_per_second` in the `_reindex` request,  `requests_per_second` can be either `-1` to turn off throttling, or any decimal number like `1.7` or `12` to throttle to that level.

Reviewer 2: ```suggestion
Similarly to setting `requests_per_second` in the `_reindex` request,  `requests_per_second` can be either `-1` to turn off throttling or any decimal number like `1.7` or `12` to throttle to that level.
```

Rethrottling that speeds up the query takes effect immediately, but rethrottling that slows down the query will take effect after completing the current batch.
This prevents the underlying search context used between batches from timing out.

> The same `scroll` versus point-in-time keep-alive rules apply, see the note under [Reindex with throttling](#docs-reindex-throttle).

Reviewer 2: ```suggestion
The same `scroll` versus point-in-time keep-alive rules apply. Refer to the note under [Reindex with throttling](#docs-reindex-throttle).
```

Adding `slices` to the reindex API just automates the manual process used in the section above, creating sub-requests which means it has some quirks:

> * {applies_to}`stack: ga` You can view these requests in the [task management APIs](https://www.elastic.co/docs/api/doc/elasticsearch/group/endpoint-tasks). These sub-requests are "child" tasks of the task for the request with `slices`.

Reviewer 2: ```suggestion
* {applies_to}`stack: ga` You can view these requests in the [task management APIs]({{es-apis}}group/endpoint-tasks). These sub-requests are "child" tasks of the task for the request with `slices`.
```
